src/runtime/runtime_external.go

```diff
@@ -0,0 +1,68 @@
+// +build machine_external
+
+package runtime
+
+type timeUnit int64
+var tickMicros = int64(1)
+var asyncScheduler = false
+
+/*
+ * Use the interface external_* below to expose to the tinygo runtime your
+ * external implementation of the runtime.  External here means outside the
+ * source tree.
+ *
+ * Example (and the //export is mandatory)
+ *
+ * //export runtime.external_putchar
+ * func putchar(c uint8) {
+ *  //your implementation goes here
+ * }
+*/
+
+//
+// external interface
+//
+
+//go:extern external_ticks()
```

Comment:
`//go:extern` in TinyGo only works for global variables, not for functions. You should use `//export` instead.